crates/notion-core/src/platform/mod.rs

```diff
@@ -8,6 +8,15 @@ use distro::node::NodeVersion;
 use notion_fail::{Fallible, ResultExt};
 use path;
 
+/// A specification of tool versions needed for a platform
+#[derive(Eq, PartialEq, Clone, Debug)]
+pub struct PlatformSpec {
+    /// The pinned version of Node.
+    pub node: NodeVersion,
```

Comment:
Is your thinking that this will get replaced by something where the npm version is optional? I was gonna say `Image` could contain a `PlatformSpec` instead of duplicating its contents, but if they're going to diverge then this is good as is.

Reply:
Yeah, I was thinking that the two types are likely to diverge, but figured that for the initial separation keeping them the same made the most sense. After #208 it would probably look something like:

```
pub struct PlatformSpec {
    pub node_runtime: Version,
    pub npm: Option<Version>,
    pub yarn: Option<Version>,
}
```

Thinking about that though, I think once that starts to happen, the `checkout` operation becomes about more than just ensuring everything is downloaded, so it might be good to bring it back and have it be `fn checkout(&self, &Session);` where it then calls out to the `Session` for the "ensure node / yarn are available" part, but then performs the other operations within the `PlatformSpec` method.

I was going to tackle #208 next, since it's all fresh in my mind, I can probably include that refactor in that work, keeping this as a minimal separation that doesn't really change the code much other than some places swapping `Image` for `PlatformSpec`